lib/ansible/modules/system/ufw.py

```diff
@@ -296,20 +297,36 @@ def ufw_version():
     (_, pre_state, _) = module.run_command(ufw_bin + ' status verbose')
     (_, pre_rules, _) = module.run_command("grep '^### tuple' /lib/ufw/user.rules /lib/ufw/user6.rules /etc/ufw/user.rules /etc/ufw/user6.rules")
 
+    changed = False
     # Execute commands
     for (command, value) in commands.items():
         cmd = [[ufw_bin], [module.check_mode, '--dry-run']]
 
         if command == 'state':
             states = {'enabled': 'enable', 'disabled': 'disable',
                       'reloaded': 'reload', 'reset': 'reset'}
-            execute(cmd + [['-f'], [states[value]]])
+            if module.check_mode:
+                if value in ['reloaded', 'reset']:
+                    changed = True
+                else:
+                    (_, ufw_state, _) = execute([[ufw_bin], ["status"]])
+                    ufw_enabled = ufw_state.find("active") != -1
+                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and ufw_enabled):
+                        changed = True
+            else:
+                execute(cmd + [['-f'], [states[value]]])
 
         elif command == 'logging':
-            execute(cmd + [[command], [value]])
+            if module.check_mode:
+                changed = True
+            else:
+                execute(cmd + [[command], [value]])
 
         elif command == 'default':
-            execute(cmd + [[command], [value], [params['direction']]])
+            if module.check_mode:
+                changed = True
```

Comment:
Same here. (`Default: deny (incoming), allow (outgoing), deny (routed)`)

Reply:
I don't know how to interpret this line. What does the second deny mean ? There's nothing in man about that.

Reply:
From the manual:
```ufw [--dry-run] default allow|deny|reject [incoming|outgoing|routed]

[...]
       default allow|deny|reject DIRECTION
              change  the  default  policy  for traffic going DIRECTION, where
              DIRECTION is one of incoming,  outgoing  or  routed.  Note  that
              existing  rules  will have to be migrated manually when changing
              the default policy. See RULE SYNTAX for more on deny and reject.
```
It means that both "directions" `incoming` and `routed` are set to `deny`, and the direction `outgoing` is set to `allow`.